src/Microsoft.AspNetCore.Routing/Matching/CandidateState.cs

```diff
@@ -41,17 +46,10 @@ internal CandidateState(Endpoint endpoint, int score)
         /// </remarks>
         public int Score { get; }
 
-        /// <summary>
-        /// Gets or sets a value which indicates where the <see cref="Http.Endpoint"/> is considered
-        /// a valid candiate for the current request. Set this value to <c>false</c> to exclude an
-        /// <see cref="Http.Endpoint"/> from consideration.
-        /// </summary>
-        public bool IsValidCandidate { get; set; }
-
         /// <summary>
         /// Gets or sets the <see cref="RouteValueDictionary"/> associated with the
         /// <see cref="Http.Endpoint"/> and the current request.
         /// </summary>
-        public RouteValueDictionary Values { get; set; }
+        public RouteValueDictionary Values { get; internal set; }
```

Comment:
So close yet so far #ReadOnlyDream

Reply:
Yeah I'm a little sad about it. I can't really think of a good solution unless we pre-allocated a `Memory<CandidateState>` and treated this as a wrapper.

If we have to tradeoff something, this seems like the best tradeoff.